src/Illuminate/Contracts/Foundation/Application.php

```diff
@@ -199,6 +200,16 @@ public function loadDeferredProviders();
      */
     public function setLocale($locale);
 
+    /**
+     * Set the current application locale temporarily, execute the given closure,
+     * then reset the locale.
+     *
+     * @param  string  $locale
+     * @param  \Closure  $withDifferentLocale
+     * @return void
+     */
+    public function withLocale($locale, Closure $withDifferentLocale);
```

Comment:
We can't add a new method to an interface on a stable release as this will break other implementations.